presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2153,6 +2153,20 @@ public void revokeTablePrivileges(ConnectorSession session, SchemaTableName sche
         return result.build();
     }
 
+    @Override
+    public void commitPartitionForCreate(ConnectorSession session, ConnectorOutputTableHandle tableHandle, Collection<Slice> fragments)
+    {
+        HiveOutputTableHandle handle = (HiveOutputTableHandle) tableHandle;
+        stagingFileCommitter.commitFiles(session, handle.getSchemaName(), handle.getTableName(), getPartitionUpdates(fragments));
+    }
+
+    @Override
+    public void commitPartitionForInsert(ConnectorSession session, ConnectorInsertTableHandle tableHandle, Collection<Slice> fragments)
```

Comment:
Since the `testInsertAndCreateTableWithWritingStagingFileEnabled` test has been removed, i would suggest adding an integration test to the `AbstractTestHiveClient` that creates a PageSink with a partition commit enabled, and then commits the results with the new SPI.